### ISSUE TITLE
Issues when Testing Bridge Connectivity and using VPN Mode

### ISSUE THREAD

**Issue by MetalNeo** [guardianproject/orbot#391]
I have noticed several issues with Orbot's bridges and Bridges menu.

Under the bridges menu, I have noticed that testing "Connect directly to Tor" always returns a positive test, even on networks that block Tor. Similarly, testing the "Community Servers" option always returns a negative test, even on networks that do not block Tor. I cannot verify if the "Cloud Servers" option is being tested correctly as I do not have access to a network that blocks it. However, I have noticed that the "Cloud Servers" is able to complete the boostrap process, but fails to actually connect to the Internet afterwards.

I have also noticed issues with VPN mode and the bridges. VPN mode works fine with no bridges selected. With "Community Servers" selected, the bootstrap process is unable to complete even though it is able to complete and connect successfully when run with VPN mode off. With "Cloud Servers" selected, the bootstrap process completes, but fails to actually connect to the Internet afterwards similarly to behavior with VPN mode off.

Version: 16.3.0-RC-1-tor-0.4.2.7 from F-Droid repo
Android Version: 9

**Comment by n8fr8** [comment]
Can you try this beta release: https://github.com/guardianproject/orbot/releases/tag/16.3.1-BETA-2-tor-0.4.3.6 ?

The Bridge Connectivity tester can sometimes be too optimistic in its connectivity testing. That is something we are working on. That said, it seems like your issue with the VPN is not necessarily related to that.

**Comment by MetalNeo** [comment]
Thank you for the link!

On open networks, all 3 options now test and bootstrap successfully. However, "Cloud Servers" still doesn't actually seem to connect to the Internet after bootstrapping.

On networks that block Tor and "Community Servers," testing Tor directly is still saying it works even though it fails to complete the bootstrap. "Community Servers" tests negative correctly and fails to bootstrap as expected. "Cloud Servers" tests successfully and finishes bootstrap, but still doesn't seem to actually be able to connect to the Internet.

As for VPN mode, both Tor direct and "Community Servers" options are working on open networks without issue now. "Cloud Servers" is testing/bootstrapping correctly, but the VPN apps are not connecting to the Tor network on both open and direct Tor blocked networks. One app was able to access the Internet without Tor, despite enabling the strict VPN system settings.

In summary, Tor is still testing postive on direct connection with blocked networks and the "Cloud Servers" bridge is still having issues establishing a connection to the Internet. There may be a potential issue with the VPN mode, but I will wait for the "Cloud Servers" bridge connectivity to be fixed first to be sure.

Please let me know if I should create separate bug reports for these remaining issues.

**Comment by bitmold** [comment]
Closing this since bridge connectivity testing has been removed
